### PR TITLE
Rename post-install script, add warnings in the files will be overwritten

### PR DESCRIPTION
**Pull request by javidhsueh**
Per our discussion in #202

### CODE REVIEW

**Comment by javidhsueh** [comment]
Or maybe we can just override the site-query.jsx for now? Before we figure out how to import it from ocular-gatsby, people still need to copy over the site-query.jsx to their folder. This post-install script can at least reduce some friction when migrating to a new version of ocular-gatsby

**Comment by ibgreen** [comment]
> Or maybe we can just copy the site-query.jsx for now?

Yes. I am also fine with copying the others IF they don't already exist.

BTW - the version of `gatsby-ssr` in ocular looks different from the one the @georgios-uber got to work. Might be worth making sure they are identical?

**Comment by javidhsueh** [comment]
> > Or maybe we can just copy the site-query.jsx for now?
> 
> Yes. I am also fine with copying the others IF they don't already exist.
> 
> BTW - the version of `gatsby-ssr` in ocular looks different from the one the @georgios-uber got to work. Might be worth making sure they are identical?

The problem is that GraphQL will error out silently without any warning due to the query we give. It happens very often that I need to copy the `site-query.jsx` from ocular repo to override my local copy. 

My point is that the post-install script should override the `site-query.jsx` no matter it exists or not.
@ibgreen

**Comment by javidhsueh** [comment]
Another question I have is that, do we need to copy `gatsby-node.js`?

**Comment by ibgreen** [comment]
> My point is that the post-install script should override the site-query.jsx no matter it exists or not.

Yes 100% agree. In case of `site-query.jsx`, it is just our hack and not part of the gatsby configuration file set. Ideally it should be `.gitignored` by the user and just copied in freshly on each install.

My point was that the other files are different, they are the documented gatsby entry points, and the user may want to edit them. But that comment does not apply to `site-query.jsx`.

Thanks for double checking!

File: modules/gatsby/bin/post-install.js [hunk 1]
```diff
@@ -19,23 +19,35 @@
...
-  console.log('Migrating', `${DIR_PATH}/${f.filename}`)
-  writeFileSync(`${DIR_PATH}/${f.filename}`, file);
+  const targetPath = `${DIR_PATH}/${f.filename}`;
+  // copy the file if it doesn't exist
+  if (!existsSync(targetPath)) {
+    const sourcePath = `${f.dir}/${f.filename}`;
+    const file = readFileSync(`${f.dir}/${f.filename}`);
+    console.log('Migrating', targetPath)
+    writeFileSync(targetPath, file);
+  }
```
Comment (georgios-uber) [inline modules/gatsby/bin/post-install.js]: Please write a message if not migrated and does not match!

**Pull request merged**